code/obj/mining_cloud_storage.dm

```diff
@@ -256,81 +256,19 @@
 			OCD.price = max(0,new_price)
 		return
 
-	attack_hand(var/mob/user as mob)
-
-		src.add_dialog(user)
-
-		if (status & BROKEN || status & NOPOWER)
-			var/dat = "The screen is blank."
-			user.Browse(dat, "window=mining_dropbox;size=400x500")
-			onclose(user, "mining_dropbox")
-			return
-
-		var/list/dat = list({"<B>[src.name]</B>
-			<br><HR>
-			<B>Rockbox™ Ore Cloud Storage Service Settings:</B>
-			<br><small>
-			<B>Rockbox™ Fees:</B> $[!rockbox_globals.rockbox_premium_purchased ? rockbox_globals.rockbox_standard_fee : 0] per ore [!rockbox_globals.rockbox_premium_purchased ? "(Purchase our Premium Service to remove this fee!)" : ""]<BR>
-			<B>Client Quartermaster Transaction Fee:</B> [rockbox_globals.rockbox_client_fee_pct]%<BR>
-			<B>Client Quartermaster Transaction Fee Per Ore Minimum:</B> $[rockbox_globals.rockbox_client_fee_min]<BR>
-			</small><HR>"})
-
-		if(ores.len)
-			for(var/ore in ores)
-				var/sellable = 0
-				var/price = 0
-				var/datum/ore_cloud_data/OCD = ores[ore]
-				price = OCD.price
-				sellable = OCD.for_sale
-				dat += "<B>[ore]:</B> [OCD.amount] (<A href='?src=\ref[src];sellable=[ore]'>[sellable ? "For Sale" : "Not For Sale"]</A>) (<A href='?src=\ref[src];price=[ore]'>$[price] per ore</A>) (<A href='?src=\ref[src];eject=[ore]'>Eject</A>)<br>"
-		else
-			dat += "No ores currently loaded.<br>"
-
-		user.Browse(dat.Join(), "window=mining_dropbox;size=500x500")
-		onclose(user, "mining_dropbox")
-
-
-
-	Topic(href, href_list)
-
-		if(status & BROKEN || status & NOPOWER)
-			return
-
-		if(usr.stat || usr.restrained())
-			return
-
-		if ((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && istype(src.loc, /turf))))
-			src.add_dialog(usr)
-
-			if (href_list["eject"])
-				var/ore = href_list["eject"]
-				src.eject_ores(ore,null,0,0,usr)
-
-			if (href_list["price"])
-				var/ore = href_list["price"]
-				var/new_price = null
-				new_price = input(usr,"What price would you like to set? (Min 0)","Set Sale Price",null) as num
-				update_ore_price(ore,new_price)
-
-			if (href_list["sellable"])
-				var/ore = href_list["sellable"]
-				update_ore_for_sale(ore)
-
-			src.updateUsrDialog()
-		return
-
 	proc/eject_ores(var/ore, var/eject_location, var/ejectamt, var/transmit = 0, var/user as mob)
 		var/amount_ejected = 0
 		if(!eject_location)
 			eject_location = get_output_location()
 		for(var/obj/item/raw_material/R in src.contents)
 			if (R.material_name == ore)
-				if (!ejectamt)
-					ejectamt = input(usr,"How many ores do you want to eject?","Eject Ores") as num
-				if ((ejectamt <= 0 || get_dist(src, user) > 1) && !transmit)
-					break
-				if (!eject_location)
-					break
+				//The new TGUI already does all of these checks, we don't need them here (and they would break it)
```

Comment:
```suggestion
				
```
Considering you got rid of the HTML code, you could probably go with deleting this instead of commenting it out

Reply:
yeah, true